crates/forge/src/compiled_raw.rs

```diff
@@ -0,0 +1,52 @@
+use cairo_lang_sierra::program::Program;
+use forge_runner::compiled_runnable::{ExpectedTestResult, FuzzerConfig, ValidatedForkConfig};
+use serde::Deserialize;
+use starknet::core::types::BlockId;
+
+#[derive(Debug, Clone, Deserialize)]
+pub(crate) struct CompiledTestCrateRaw {
+    pub sierra_program: Program,
```

Comment:
In `scarb-snforge-test-collector` `VersionedProgram` is used. This works anyway, but the fact that `Program` deserializes from `VersionedProgram` is an effect of custom serialization, we are just lucky here.

To change Program to VersionedProgram we need to bump Cairo to main or wait for 2.4.0, since the serialization of VersionedProgram on 2.3.1 is different then that on main.

cc @cptartur @MaksymilianDemitraszek @maciektr

Reply:
Will wait since we will prob need to revert the serialization changes in the compiler to keep cast backwards compatible, then I could just use another nightly (with reverted serialization) and it would work